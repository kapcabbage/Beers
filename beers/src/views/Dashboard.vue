<template>
  <v-container 
  fluid 
  fill-height>
    <v-layout row wrap>
      <v-flex xs12>
        <h2 class="font-weight-thin display-3">Look at all this beers</h2>
      </v-flex>
       <v-flex xs12>
          <v-text-field
            solo
            light
            round
            clearable
            label="Search beer name"
            append-icon="search"
            v-on:keyup.enter="search"
            @click:clear="clear"
            v-model="searchName"
          ></v-text-field>
        </v-flex>
      <v-container fluid grid-list-md>
        <v-layout row wrap justify-start>
          <beer-tile v-for="(beer, index) in beers" :key="index" :beer="beer"/>
            
        <v-flex xs12 v-if="load_vis" class="last"> 
          <v-layout flex row justify-space-around>       
            <v-btn outline round v-on:click="loadMore" color="grey lighten-1">Load more</v-btn>
          </v-layout>
        </v-flex>
        </v-layout>
      </v-container>
      
    </v-layout>
  </v-container>
</template>

<script>
import { LOAD_BEERS, LOAD_MORE_BEERS, SEARCH_BEER_NAME} from "../stores/actions.type";
import { GET_BEERS, GET_CURRENTPAGE, GET_TOTALPAGE } from "../stores/getters.type";
import { mapGetters } from "vuex";
import BeerTile from '../components/BeerTile.vue';

export default {
  data() {
    return {
      visibility : false,
      searchName: "",
      show: false
    };
  },
  components: {
    BeerTile
  },
  mounted() {
    this.$store.dispatch(LOAD_BEERS);
  },
  methods: {
   loadMore(){
     this.$store.dispatch(LOAD_MORE_BEERS, {pageNumber: (this.currentPage + 1)});
   },
   search(){
     console.log(this.searchName)
     if(this.searchName === ""){
       this.$store.dispatch(LOAD_BEERS);
     }
     else{
       this.$store.dispatch(SEARCH_BEER_NAME, {name: this.searchName});
     }
     
   },
   clear(){
       this.$store.dispatch(LOAD_BEERS);
   }
  },

  computed: {
    ...mapGetters({
      beers: GET_BEERS,
      currentPage : GET_CURRENTPAGE,
      totalPage: GET_TOTALPAGE
    }),

    load_vis: function(){
        if(this.beers.lenght === 0){
          return false;
        }
        if(this.currentPage === this.totalPage){
          return false;
        }
        return true
    },
  }
};
</script>

<style  lang="scss">
  .last {
    order: 3;
  }
</style>