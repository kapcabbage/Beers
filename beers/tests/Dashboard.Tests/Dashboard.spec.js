import {shallowMount,createLocalVue} from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Vue from 'vue';
import Dashboard from '../../src/views/Dashboard.vue';
import dashboardModule from '../../src/stores/dashboard.module';
import moment from 'moment';

const localVue = createLocalVue()

localVue.use(Vuex)
Vue.use(Vuetify);
describe('Dashboard.vue', () => {
    let store;
    let dashboard;
    let actions;

    beforeEach(() => {
        actions = {

        }
        dashboard = {
            state:{
              
            },
            actions:{
               
            },
            getters:{
               
            }
        }
        store = new Vuex.Store({
            modules: {
                dashboardModule
            }
            
        });
    });
    it('Does *Dashboard* page render correctly', () => {
        const wrapper = shallowMount(Dashboard,{store, localVue})
        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    
})