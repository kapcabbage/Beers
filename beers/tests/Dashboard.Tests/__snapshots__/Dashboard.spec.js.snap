// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue Does *Dashboard* page render correctly 1`] = `
<v-container-stub
  fill-height=""
  fluid=""
  grid-list-xl=""
  tag="div"
>
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      tag="div"
      xs12=""
    >
      <h2
        class="font-weight-thin display-3"
      >
        Dashboard
      </h2>
    </v-flex-stub>
     
    <v-flex-stub
      tag="div"
      xs12=""
    >
      <v-layout-stub
        fill-height=""
        tag="div"
        wrap=""
      >
        <v-flex-stub
          tag="div"
          xs12=""
        >
          <calendar-card-stub
            eventsmap="[object Object]"
            today="2019-04-04"
          >
            <h4
              class="title font-weight-light"
            >
              Calendar
            </h4>
             
            <p
              class="category d-inline-flex font-weight-light"
            >
              <v-icon-stub
                color="green"
                small=""
              >
                
							-arrow-up
							
              </v-icon-stub>
               
              <span
                class="green--text"
              >
                Available
              </span>
              Â 
							spots overview
						
            </p>
             
            <template>
              <v-icon-stub
                class="mr-2"
                small=""
              >
                
							mdi-clock-outline
							
              </v-icon-stub>
               
              <span
                class="caption grey--text font-weight-light"
              >
                updated 4 minutes ago
              </span>
            </template>
          </calendar-card-stub>
        </v-flex-stub>
      </v-layout-stub>
    </v-flex-stub>
     
    <v-flex-stub
      sm4=""
      tag="div"
      xs12=""
    >
      <card-stub
        color="info"
        elevation="10"
        icon="directions_car"
        offset="24"
        subicon="update"
        subtext="Just Updated"
        title="Available spots/Month"
        value="+245"
      />
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
